### PR TITLE
Enable internal Artifactory server

### PR DESCRIPTION
**Pull request by beckerhe**
This PR enables the internal Artifactory server if reachable.

The bootstrap scripts now call a script called `contrib/conan/configs/install.{ps1,sh}`.
This script takes care of installing the conan configuration. That involves the following points:
- Install the static windows or linux configuration depending on the platform.
- Detect whether the internal artifactory server is reachable (either because it's a CI server or it's an internal workstation) and change the remote servers accordingly. If the internal artifactory is available public remotes will be disabled and only the internal package remote server will be used.

Further changes: 
- Added License headers to all the involved scripts
- The CI build scripts also call the `install.{ps1,sh}`-script.
- `bootstrap-orbit-ggp.{sh,ps1,bat}` is now deprecated because it is not needed anymore since we have the `ggp_sdk` on our internal artifactory server. So no special treatment for a GGP build is necessary anymore.

### CODE REVIEW

File: bootstrap-orbit-ggp.sh [hunk 1]
```diff
@@ -1,26 +1,9 @@
...
-  if [ ! -d $DIR/contrib/conan/recipes/ggp_sdk ]; then
-    git clone sso://user/hebecker/conan-ggp_sdk $DIR/contrib/conan/recipes/ggp_sdk || exit $?
-  fi
-  conan export $DIR/contrib/conan/recipes/ggp_sdk orbitdeps/stable || exit $?
-else
-  echo "ggp_sdk seems to be installed already. Skipping installation step..."
-fi
-
-$DIR/build.sh ggp_release
+exec "${DIR}/bootstrap-orbit.sh" ggp_release
```
Comment (danielfenner) [inline bootstrap-orbit-ggp.sh:9]: Doesn't really matter that much but you could add a deprecation message here as well.

File: contrib/conan/configs/install.ps1 [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      $process = Start-Process $conan.Path -Wait -NoNewWindow -ErrorAction Stop -PassThru -ArgumentList "remote","add","-i","0","-f","artifactory","http://$location/artifactory/api/conan/conan"
+      if ($process.ExitCode -ne 0) { Throw "Error while adding conan remote." }
+
+      $process = Start-Process $conan.Path -Wait -NoNewWindow -ErrorAction Stop -PassThru -ArgumentList "config","set","proxies.http=`"$location=http://127.0.0.2:999`""
+      if ($process.ExitCode -ne 0) { Throw "Error while adding proxy setting to conan." }
+
+      conan_disable_public_remotes
+
+    } catch {
+      # Nothing to do in this case
```
Comment (danielfenner) [inline contrib/conan/configs/install.ps1:50]: Maybe something like 
Write-Host "Using public remotes."

File: contrib/conan/configs/install.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  conan remote disable bincrafters || return $?
+  return 0
+}
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+if [ "$(uname -s)" == "Linux" ]; then
+  conan config install "$DIR/linux" || exit $?
+else
+  conan config install "$DIR/windows" || exit $?
```
Comment (danielfenner) [inline contrib/conan/configs/install.sh:19]: Does this ever get used?

File: contrib/conan/configs/install.ps1 [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+function conan_disable_public_remotes() {
+  $remotes = "bintray", "conan-center", "bincrafters"
+  foreach ($remote in $remotes) {
+    $process = Start-Process $conan.Path -Wait -NoNewWindow -ErrorAction Stop -PassThru -ArgumentList "remote","disable",$remote
+    if ($process.ExitCode -ne 0) { Throw "Error while disabling conan-remote $remote." }
+  }
+}
+
+$process = Start-Process $conan.Path -Wait -NoNewWindow -ErrorAction Stop -PassThru -ArgumentList "config","install","$PSScriptRoot\windows"
```
Comment (danielfenner) [inline contrib/conan/configs/install.ps1:15]: Myunderstanding is that this line actives the public remotes. Where is this configured? Maybe add a comment.

File: bootstrap-orbit-ggp.sh (hunk 1 above)
Comment (beckerhe) [reply bootstrap-orbit-ggp.sh]: Yes, definitely. Thanks for spotting this! Done.

File: contrib/conan/configs/install.ps1 (hunk 2 above)
Comment (beckerhe) [reply contrib/conan/configs/install.ps1]: Good point. Done!

File: contrib/conan/configs/install.sh (hunk 3 above)
Comment (beckerhe) [reply contrib/conan/configs/install.sh:20]: Yes, I use that quite a lot from git-bash. It's not integrated yet but I want to call this from my `build_and_upload_dependencies.sh` script which also works on Windows.

File: contrib/conan/configs/install.ps1 (hunk 4 above)
Comment (beckerhe) [reply contrib/conan/configs/install.ps1:16]: Comment added. +1

**Comment by pierricgimmig** [comment]
@beckerhe , LGTM, thanks.  Please update internal doc with new build instructions for ggp.

**Pull request merged**